docs/4.2/production.md

```diff
@@ -0,0 +1,269 @@
+
+
+```yaml
+[Unit]
+Description=Teleport SSH Service
+After=network.target
+
+[Service]
+Type=simple
+Restart=on-failure
+# Set the nodes roles with the `--roles`
+# In most production environments you will not
+# want to run all three roles on a single host
+# --roles='proxy,auth,node' is the default value
+# if none is set
+ExecStart=/usr/local/bin/teleport start --roles=auth --config=/etc/teleport.yaml --pid-file=/var/run/teleport.pid
+ExecReload=/bin/kill -HUP $MAINPID
+PIDFile=/var/run/teleport.pid
+
+[Install]
+WantedBy=multi-user.target
+```
+
+There are a couple of important things to notice about this file:
+
+1. The start command in the unit file specifies `--config` as a file and there
+   are very few flags passed to the `teleport` binary. Most of the configuration
+   for Teleport should be done in the [configuration file](admin-guide.md#configuration).
+
+2. The **ExecReload** command allows admins to run `systemctl reload teleport`.
+   This will attempt to perform a graceful restart of _*but it only works if
+   network-based backend storage like [DynamoDB](../configuration/#storage) or
+   [etc 3.3](../configuration/#storage) is configured*_. Graceful Restarts will
+   fork a new process to handle new incoming requests and leave the old daemon
+   process running until existing clients disconnect.
+
+### Start the Teleport Service
+
+You can start Teleport as a Systemd Unit by enabling the `.service` file
+with the `systemctl` tool.
+
+```bash
+$ cd /etc/systemd/system
+# Use your text editor of choice to create the .service file
+# Here we use vim
+$ vi teleport.service
+# use the file above as is, or customize as needed
+# save the file
+$ systemctl enable teleport
+$ systemctl start teleport
+# show the status of the unit
+$ systecmtl status teleport
+# follow tail of service logs
+$ journalctl -fu teleport
+# If you modify teleport.service later you will need to
+# reload the systemctl daemon and reload teleport
+# to apply your changes
+$ systemctl daemon-reload
+$ systemctl reload teleport
+```
+
+You can also perform restarts or upgrades by sending `kill` signals
+to a Teleport daemon manually.
+
+| Signal                  | Teleport Daemon Behavior
+|-------------------------|---------------------------------------
+| `USR1`                  | Dumps diagnostics/debugging information into syslog.
+| `TERM`, `INT` or `KILL` | Immediate non-graceful shutdown. All existing connections will be dropped.
+| `USR2`                  | Forks a new Teleport daemon to serve new connections.
+| `HUP`                   | Forks a new Teleport daemon to serve new connections **and** initiates the graceful shutdown of the existing process when there are no more clients connected to it. This is the signal sent to trigger a graceful restart.
```

Comment:
This information is also specified somewhere else in the docs. Again it would be nice to keep it all in one place so we don't duplicate things and then forget to update in future. Not ideal perhaps but I'm unsure of a better way to solve this.

Reply:
Yeah, I agree. Some of are repetitive. There was a thought to split key sections and include them but it would require a few more tweaks to MkDocs, but it is something I would like to do.